packages/utils/src/modules-sdk/decorators/inject-transaction-manager.ts

```diff
@@ -1,7 +1,10 @@
 import { Context, SharedContext } from "@medusajs/types"
+import { isString } from "../../common"
 
 export function InjectTransactionManager(
-  shouldForceTransaction: (target: any) => boolean = () => false,
+  shouldForceTransactionOrManagerProperty:
```

Comment:
question: Can you explain this change?

In the diff, I can see a lot of changes from:
```
@InjectTransactionManager("productTagRepository_")
```
to 
```
@InjectTransactionManager("productTagRepository_")
```

But just as many changed from:
```
@InjectTransactionManager(doNotForceTransaction, "productTagRepository_")
```
to
```
@InjectTransactionManager("productTagRepository_")
```

How does that work?

Reply:
Have we been using this incorrectly until now?

Reply:
It was a legacy, so basically what we were doing was that we were forcing to create a transaction only when a module is isolated. But the reallity is that every time you are doing a write operation it needs to happen in a transaction right? Now we don't pass this first function, which means if the transaction manager is present then we are calling the next task and propagating the sharedContext. So it was more of a legacy from inventory/stock location which use another decorator which was doing the same thing. Hope it make sense 😂

Reply:
Got it. This is probably also relevant for @pKorsholm @riqwan

Reply:
> But the reallity is that every time you are doing a write operation it needs to happen in a transaction right

If the modules share resources, shouldn't we reuse the transaction throughout the entire flow?

Reply:
If it share resources, it means that it is sharing the connection which is used by mikro orm to init a new instance of itself. If you want to span a transaction across modules then you can pass it as part of the context. But sharing the transaction and manager is something that pnly occurs for the inventory and stock location modules. Product and pricing have there own mikro orm instance on shared resources

Reply:
In a flow, each step is independent and compensated independently

Reply:
The transaction sharing is only something used on inventory and stock-location, where the module is using the same typeorm transaction of core.
This is not valid for new modules.

Reply:
> every time you are doing a write operation it needs to happen in a transaction right

This is clear, but maybe I misunderstood this one. Does this mean that if you are using two modules e.g. in an endpoint, not in a workflow, the consumer will have to manage a potential shared transaction between the two modules via the context param?

Reply:
Only if there are using mikro orm around to wrap, otherwise there are in two different transactions as it is today. Today it is forcing a transaction which does the same as not forcing it since there is no transactionManager pass to any of the modules from the end point. If one succeed and the other one fail, we have to handle the compensation ourselves, this is why we move to workflow for the wrote end point (but all of them could be workflow)

Reply:
Got it thx - makes sense!

Reply:
If you see a mistake don't hesitate to share it in case i ve missed something